providers/jaxb/src/main/java/org/jboss/resteasy/plugins/providers/jaxb/JAXBContextWrapper.java

```diff
@@ -81,11 +105,32 @@ public JAXBContextWrapper(JAXBContext wrappedContext, JAXBConfig config) throws
     * @param config
     * @throws JAXBException
     */
-   public JAXBContextWrapper(Class<?>[] classes, Map<String, Object> properties, JAXBConfig config)
+   public JAXBContextWrapper(final Class<?>[] classes, final Map<String, Object> properties, JAXBConfig config)
            throws JAXBException
    {
       processConfig(config);
-      wrappedContext = JAXBContext.newInstance(classes, properties);
+      try
+      {
+         if (System.getSecurityManager() == null)
+         {
+            wrappedContext = JAXBContext.newInstance(classes, properties);
+         }
+         else
+         {
+            wrappedContext = AccessController.doPrivileged(new PrivilegedExceptionAction<JAXBContext>()
+            {
+               @Override
+               public JAXBContext run() throws JAXBException
+               {
+                  return JAXBContext.newInstance(classes, properties);
+               }
+            });
+         }
+      }
+      catch (PrivilegedActionException paex)
+      {
+         throw new JAXBException(paex.getMessage());

```

Comment:
I would suggest to change this to JAXBException(paex.getMessage(), paex) , to keep stacktrace of the original exception.